R/utils-get_code_dependency.R

```diff
@@ -0,0 +1,498 @@
+# get_code_dependency ----
+
+#' Return the lines of code (with side-effects) needed to reproduce the object
+#'
+#' @details This function assumes that object relationships are established using the `<-`, `=`, or `->` assignment
+#' operators. It does not support other object creation methods like `assign` or `<<-`, nor non-standard-evaluation
+#' methods. To specify relationships between side-effects and objects, you can use the comment tag
+#' `# @linksto object_name` at the end of a line where the side-effect occurs.
+#'
+#' @param code An `expression` with `srcref` attribute or a `character` with the code.
+#' @param names A `character(n)` with object names.
+#'
+#' @return `character` vector of elements of `code` calls that were required to build the side-effects and
+#' influencing objects having and impact on the `object`
+#'
+#' @keywords internal
+get_code_dependency <- function(code, names) {
+  assert_classes(code, names)
+
+  if (is_empty(code)) return(code)
```

Comment:
function used only once, don't need to create sugar in my opinion

```suggestion
  if (identical(code, character(0)) || identical(trimws(code), "")) return(code)
```

Reply:
I don't know whether this is easier. I needed to scroll down and figure out what `is_empty` means. And it means something else than `rlang` equivalent.

```r
rlang::is_empty("")
# [1] FALSE
```

Reply:
I do select the name and press ctrl/fn+F2 in RStudio and it redirects you to the function definition. IMHO there is less clutter reviewing and you can focus on major parts of the implementation during the review and do not necessarily need to look at assertions that are just long. For sure I will brought this back to `get_code_dependencies`

Reply:
> IMHO there is less clutter reviewing and you can focus on major parts

Yes, but you have also extra code to maintain the package. I understand the reason but I disagree. I think you made a great job by splitting former functions into meaningful parts but these utilities is little too much in my opinion. When analysing the code you need to know what each line does. Here, instead of knowing "what is the assertion on the code" one needs to go to the private util to read two lines of code. So not a great simplification of the code - one vs two.

`assert_classes` is wrong for many reasons. Name is wrong, as it looks more like `assert_get_code_dependency_formals`. `assert_classes(code, names)` sounds like `checkmate::assert_class` but it is something completely different. It doesn't fit to any `assert_` function.

I've already highlighted that `is_empty` is exactly the same name as `rlang::is_empty` function but return different results. `rlang` is by the way used in the package so I had reasonable suspicious that this is `rlang` function.

Reply:
Yes, I see it didnt bring any help. I did mention in my first comment, when asking for a review, that I am not planning to keep those ever https://github.com/insightsengineering/teal.data/pull/201#issuecomment-1827990169 Will remove those utils as it only causes unnecessary confusion!

Reply:
Sorry for the confusion - just removed 7402a4dc752cda2e6e7c79f0f8f32f8c78d2cf02 !